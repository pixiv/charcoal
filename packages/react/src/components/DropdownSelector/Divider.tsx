import styled from 'styled-components'

/**
 * 水平方向の直線
 */
export const Divider = styled.div.attrs({ role: 'separator' })`
  display: flex;

  &:before {
    content: '';
    display: block;
    width: 100%;
    height: 1px;
    background: #00000014;
  }
`
