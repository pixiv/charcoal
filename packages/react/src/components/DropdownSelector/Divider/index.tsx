import { JSX } from 'react/jsx-runtime'
import './index.css'

export function Divider(): JSX.Element {
  return <div role="separator" className="charcoal-menu-group-divider" />
}
