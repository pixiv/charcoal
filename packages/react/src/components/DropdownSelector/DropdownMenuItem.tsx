import styled from 'styled-components'
import MenuItem, { MenuItemProps } from './MenuItem'
import { MenuListContext } from './MenuList/MenuListContext'
import React, { useContext } from 'react'
import { theme } from '../../styled'
import Icon from '../Icon'

export type DropdownMenuItemProps = Omit<MenuItemProps<'div'>, 'as'>

/**
 * DropdownSelectorの選択肢として使うMenuItem
 */
export default function DropdownMenuItem(props: DropdownMenuItemProps) {
  const { value: ctxValue } = useContext(MenuListContext)
  const isSelected = props.value === ctxValue
  const { children, ...rest } = props

  return (
    <MenuItem {...rest}>
      {isSelected && <Text2ColorIcon name="16/Check" />}
      <StyledSpan isSelected={isSelected}>{props.children}</StyledSpan>
    </MenuItem>
  )
}

/**
 * アイコンがない時を考慮して20px（16pxのwidthと4pxのgap）の余白をとる
 */
const StyledSpan = styled.span<{ isSelected?: boolean }>`
  ${theme((o) => [o.typography(14), o.font.text2])};
  display: flex;
  align-items: center;
  width: 100%;
  margin-left: ${({ isSelected }) => (isSelected === true ? 0 : 20)}px;
`

const Text2ColorIcon = styled(Icon)`
  ${theme((o) => [o.font.text2])}
  padding-right: 4px;
`
