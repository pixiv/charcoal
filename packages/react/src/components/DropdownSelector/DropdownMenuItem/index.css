.charcoal-dropdown-selector-menu-item {
  font-size: 14px;
  line-height: 22px;
  color: var(--charcoal-color-text-secondary-default);
  padding: 9px 0;

  display: flex;
  align-items: center;
  width: 100%;
  margin-left: 20px;
}

.charcoal-dropdown-selector-menu-item[data-selected='true'] {
  margin-left: 0px;
}

.charcoal-dropdown-selector-menu-item-icon {
  color: var(--charcoal-color-text-secondary-default);
  padding-right: 4px;
}
