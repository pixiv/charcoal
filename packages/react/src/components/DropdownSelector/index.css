.charcoal-dropdown-selector-root {
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 4px;
  width: 100%;
  border-radius: 4px;
}

.charcoal-dropdown-selector-root[aria-disabled='true'] {
  cursor: default;
  opacity: 0.32;
}

.charcoal-dropdown-selector-button {
  display: grid;
  grid-template-columns: 1fr auto;
  justify-content: space-between;
  align-items: center;

  height: 40px;
  width: 100%;
  box-sizing: border-box;
  border: none;
  cursor: pointer;
  gap: 4px;

  padding-right: 8px;
  padding-left: 8px;
  background-color: var(--charcoal-color-container-secondary-default-a);
  border-radius: 4px;

  transition: 0.2s box-shadow, 0.2s background-color;
}

.charcoal-dropdown-selector-button:disabled {
  cursor: default;
}

.charcoal-dropdown-selector-button:not(:disabled):focus {
  outline: none;
  box-shadow: var(--charcoal-color-custom-focus-effect);
}

.charcoal-dropdown-selector-button:not(:disabled):focus-visible {
  box-shadow: var(--charcoal-color-custom-focus-effect);
}

.charcoal-dropdown-selector-button:not(:disabled)[data-active='true'],
.charcoal-dropdown-selector-button:not(:disabled):active {
  background-color: var(--charcoal-color-container-secondary-press-a);
}

.charcoal-dropdown-selector-button:not(:disabled):hover {
  background-color: var(--charcoal-color-container-secondary-hover-a);
}

.charcoal-dropdown-selector-button:not(:disabled):focus:not(:focus-visible) {
  box-shadow: none;
}

.charcoal-dropdown-selector-button[aria-invalid='true'],
.charcoal-dropdown-selector-button:not(
    :disabled
  )[aria-invalid='true']:focus:not(:focus-visible) {
  box-shadow: var(--charcoal-color-custom-error-effect);
}

.charcoal-ui-dropdown-selector-text {
  text-align: left;
  font-size: 14px;
  line-height: 22px;
  display: flow-root;
  color: var(--charcoal-color-text-default);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.charcoal-ui-dropdown-selector-text[data-placeholder='true'] {
  color: var(--charcoal-color-text-placeholder-default);
}

.charcoal-ui-dropdown-selector-icon {
  color: var(--charcoal-color-icon-secondary-default);
}
