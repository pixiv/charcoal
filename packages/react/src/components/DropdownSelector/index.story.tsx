import { action } from '@storybook/addon-actions'
import React, { useState } from 'react'
import { Section } from 'react-stately'
import DropdownSelector, {
  DropdownSelectorItem,
  DropdownSelectorProps,
} from '.'
import { Story } from '../../_lib/compat'
import Clickable from '../Clickable'

export default {
  title: 'DropdownSelector',
  component: DropdownSelector,
}

type Props = Omit<
  DropdownSelectorProps,
  'subLabel' | 'children' | 'onOpenChange'
>
export const Default: Story<Props> = (props) => {
  return (
    <div style={{ width: 288 }}>
      <DropdownSelector
        {...props}
        placeholder={props.placeholder ?? 'Drop Down menu'}
        onChange={action('change')}
        onOpenChange={action('open')}
      >
        <DropdownSelectorItem key="k:1">選択肢1</DropdownSelectorItem>
        <DropdownSelectorItem key="k:2">選択肢2</DropdownSelectorItem>
        <DropdownSelectorItem key="k:3">選択肢3</DropdownSelectorItem>
      </DropdownSelector>
    </div>
  )
}
Default.args = {
  label: 'Label',
  requiredText: '*必須',
  required: false,
  showLabel: false,
  invalid: false,
  disabled: false,
}

export const Sections: Story<DropdownSelectorProps> = (props) => {
  return (
    <div>
      <DropdownSelector
        {...props}
        placeholder={'Drop Down menu'}
        onChange={action('change')}
        onOpenChange={action('open')}
      >
        <Section title="Section1">
          <DropdownSelectorItem key="1">選択肢1</DropdownSelectorItem>
        </Section>
        <Section title="Section2">
          <DropdownSelectorItem key="2">選択肢2</DropdownSelectorItem>
          <DropdownSelectorItem key="3">選択肢3</DropdownSelectorItem>
        </Section>
      </DropdownSelector>
    </div>
  )
}

export const Bottom: Story<DropdownSelectorProps> = (props) => {
  return (
    <div style={{ marginTop: '1000px' }}>
      <DropdownSelector
        {...props}
        placeholder={'Drop Down menu'}
        onChange={action('change')}
        onOpenChange={action('open')}
      >
        <DropdownSelectorItem key="1">選択肢1</DropdownSelectorItem>
        <DropdownSelectorItem key="2">選択肢2</DropdownSelectorItem>
        <DropdownSelectorItem key="3">選択肢3</DropdownSelectorItem>
      </DropdownSelector>
    </div>
  )
}

export const Many: Story<DropdownSelectorProps> = (props) => {
  const [value, setValue] = useState('50')
  return (
    <div style={{ padding: '300px 100px' }}>
      <DropdownSelector
        {...props}
        placeholder={'Drop Down menu'}
        onChange={(v) => {
          setValue(v.toString())
          action('change')
        }}
        onOpenChange={action('open')}
        value={value}
      >
        {[...(Array(100) as undefined[])].map((_, i) => {
          return (
            <DropdownSelectorItem textValue={i.toString()} key={i}>
              選択肢{i}
            </DropdownSelectorItem>
          )
        })}
      </DropdownSelector>
    </div>
  )
}

type HasLabelProps = {
  disabled?: boolean
}
export const HasLabel: Story<HasLabelProps> = ({ disabled }) => {
  const defaultProps: Omit<DropdownSelectorProps, 'children'> = {
    required: true,
    showLabel: true,
    label: 'Label',
    requiredText: '*必須',
    subLabel: <Clickable onClick={action('label-click')}>Text Link</Clickable>,
    assertiveText: 'Hint',
  }
  return (
    <div style={{ width: 288 }}>
      <DropdownSelector
        {...defaultProps}
        disabled={disabled}
        placeholder={'Drop Down menu'}
        onChange={action('change')}
        onOpenChange={action('open')}
      >
        <DropdownSelectorItem key="1">選択肢1</DropdownSelectorItem>
        <DropdownSelectorItem key="2">選択肢2</DropdownSelectorItem>
        <DropdownSelectorItem key="3">選択肢3</DropdownSelectorItem>
      </DropdownSelector>
    </div>
  )
}

HasLabel.args = {
  disabled: false,
}

type WithSeparatorProps = {
  mode: 'default' | 'separator'
}
export const WithSeparator: Story<WithSeparatorProps> = ({
  mode,
  ...props
}) => {
  const defaultProps: Omit<DropdownSelectorProps, 'children'> = {
    required: true,
    showLabel: true,
    label: 'Label',
    requiredText: '*必須',
    subLabel: <Clickable onClick={action('label-click')}>Text Link</Clickable>,
    assertiveText: 'Hint',
  }
  return (
    <div style={{ width: 288 }}>
      <DropdownSelector
        {...defaultProps}
        mode={mode}
        placeholder={'Drop Down menu'}
        onChange={action('change')}
        onOpenChange={action('open')}
        {...props}
      >
        <DropdownSelectorItem key="1">選択肢1</DropdownSelectorItem>
        <DropdownSelectorItem key="2">選択肢2</DropdownSelectorItem>
        <DropdownSelectorItem key="3">選択肢3</DropdownSelectorItem>
      </DropdownSelector>
    </div>
  )
}

WithSeparator.args = {
  mode: 'separator',
}

type InvalidProps = {
  disabled?: boolean
}
export const Invalid: Story<InvalidProps> = ({ disabled }) => {
  const props: Omit<DropdownSelectorProps, 'children'> = {
    label: '',
    assertiveText: 'error message',
    invalid: true,
  }
  return (
    <div style={{ width: 288 }}>
      <DropdownSelector
        {...props}
        disabled={disabled}
        placeholder={'Drop Down menu'}
        onChange={action('change')}
        onOpenChange={action('open')}
      >
        <DropdownSelectorItem key="1">選択肢1</DropdownSelectorItem>
        <DropdownSelectorItem key="2">選択肢2</DropdownSelectorItem>
        <DropdownSelectorItem key="3">選択肢3</DropdownSelectorItem>
      </DropdownSelector>
    </div>
  )
}

Invalid.args = {
  disabled: false,
}
