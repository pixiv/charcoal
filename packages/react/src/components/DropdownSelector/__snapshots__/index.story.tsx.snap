// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots DropdownSelector Basic 1`] = `
.c0 {
  display: inline-block;
  width: 100%;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
  opacity: 0.32;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  width: 100%;
  box-sizing: border-box;
  border: none;
  cursor: pointer;
  padding-right: 8px;
  padding-left: 8px;
  background-color: var(--charcoal-surface3);
  border-radius: 4px;
  -webkit-transition: 0.2s box-shadow;
  transition: 0.2s box-shadow;
}

.c1:disabled,
.c1[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c1:not(:disabled):not([aria-disabled]):focus,
.c1[aria-disabled='false']:focus,
.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active,
.c1:not(:disabled):not([aria-disabled]):focus-visible,
.c1[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c2 {
  text-align: left;
  font-size: 14px;
  line-height: 22px;
  display: flow-root;
  color: var(--charcoal-text2);
}

.c2::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c2::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c3 {
  color: var(--charcoal-text2);
}

<div
  data-dark={false}
>
  <div
    style={
      Object {
        "width": 288,
      }
    }
  >
    <div
      className="c0"
    >
      <button
        className="c1"
        onClick={[Function]}
        type="button"
      >
        <span
          className="c2"
        >
          Apple
        </span>
        <pixiv-icon
          class="c3"
          name="16/Menu"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots DropdownSelector Custom Children 1`] = `
.c0 {
  display: inline-block;
  width: 100%;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
  opacity: 0.32;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  width: 100%;
  box-sizing: border-box;
  border: none;
  cursor: pointer;
  padding-right: 8px;
  padding-left: 8px;
  background-color: var(--charcoal-surface3);
  border-radius: 4px;
  -webkit-transition: 0.2s box-shadow;
  transition: 0.2s box-shadow;
}

.c1:disabled,
.c1[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c1:not(:disabled):not([aria-disabled]):focus,
.c1[aria-disabled='false']:focus,
.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active,
.c1:not(:disabled):not([aria-disabled]):focus-visible,
.c1[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c2 {
  text-align: left;
  font-size: 14px;
  line-height: 22px;
  display: flow-root;
  color: var(--charcoal-text2);
}

.c2::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c2::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c3 {
  color: var(--charcoal-text2);
}

<div
  data-dark={false}
>
  <div
    style={
      Object {
        "width": 288,
      }
    }
  >
    <div
      className="c0"
    >
      <button
        className="c1"
        onClick={[Function]}
        type="button"
      >
        <span
          className="c2"
        >
          <div
            style={
              Object {
                "color": "pink",
                "fontWeight": "bold",
              }
            }
          >
            Apple
          </div>
        </span>
        <pixiv-icon
          class="c3"
          name="16/Menu"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots DropdownSelector In Form Tag 1`] = `
.c0 {
  display: inline-block;
  width: 100%;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
  opacity: 0.32;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  width: 100%;
  box-sizing: border-box;
  border: none;
  cursor: pointer;
  padding-right: 8px;
  padding-left: 8px;
  background-color: var(--charcoal-surface3);
  border-radius: 4px;
  -webkit-transition: 0.2s box-shadow;
  transition: 0.2s box-shadow;
}

.c1:disabled,
.c1[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c1:not(:disabled):not([aria-disabled]):focus,
.c1[aria-disabled='false']:focus,
.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active,
.c1:not(:disabled):not([aria-disabled]):focus-visible,
.c1[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c2 {
  text-align: left;
  font-size: 14px;
  line-height: 22px;
  display: flow-root;
  color: var(--charcoal-text2);
}

.c2::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c2::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c3 {
  color: var(--charcoal-text2);
}

<div
  data-dark={false}
>
  <form
    style={
      Object {
        "width": 288,
      }
    }
  >
    <div
      className="c0"
    >
      <button
        className="c1"
        onClick={[Function]}
        type="button"
      >
        <span
          className="c2"
        />
        <pixiv-icon
          class="c3"
          name="16/Menu"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`Storyshots DropdownSelector In Modal 1`] = `
.c11 {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  padding: 0;
  border-style: none;
  outline: none;
  color: inherit;
  text-rendering: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  word-spacing: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: inherit;
  margin: 0;
  overflow: visible;
  text-transform: none;
}

.c11:disabled,
.c11[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c11:focus {
  outline: none;
}

.c11::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c7 {
  display: inline-block;
  width: 100%;
}

.c7:disabled,
.c7[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
  opacity: 0.32;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  width: 100%;
  box-sizing: border-box;
  border: none;
  cursor: pointer;
  padding-right: 8px;
  padding-left: 8px;
  background-color: var(--charcoal-surface3);
  border-radius: 4px;
  -webkit-transition: 0.2s box-shadow;
  transition: 0.2s box-shadow;
}

.c8:disabled,
.c8[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c8:not(:disabled):not([aria-disabled]):focus,
.c8[aria-disabled='false']:focus,
.c8:not(:disabled):not([aria-disabled]):active,
.c8[aria-disabled='false']:active,
.c8:not(:disabled):not([aria-disabled]):focus-visible,
.c8[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c9 {
  text-align: left;
  font-size: 14px;
  line-height: 22px;
  display: flow-root;
  color: var(--charcoal-text2);
}

.c9::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c9::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c10 {
  color: var(--charcoal-text2);
}

.c12 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 32px;
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: var(--charcoal-text3);
  background-color: var(--charcoal-transparent);
  border-radius: 999999px;
  -webkit-transition: 0.2s background-color,0.2s box-shadow;
  transition: 0.2s background-color,0.2s box-shadow;
}

.c12:not(:disabled):not([aria-disabled]):hover,
.c12[aria-disabled='false']:hover {
  background-color: var( --charcoal-transparent-hover );
}

.c12:not(:disabled):not([aria-disabled]):active,
.c12[aria-disabled='false']:active {
  background-color: var( --charcoal-transparent-press );
}

.c12:not(:disabled):not([aria-disabled]):focus,
.c12[aria-disabled='false']:focus,
.c12:not(:disabled):not([aria-disabled]):active,
.c12[aria-disabled='false']:active,
.c12:not(:disabled):not([aria-disabled]):focus-visible,
.c12[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c12:disabled,
.c12[aria-disabled]:not([aria-disabled='false']) {
  opacity: 0.32;
}

.c1 {
  margin: auto;
  position: relative;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  width: 440px;
  background-color: var(--charcoal-surface1);
  border-radius: 24px;
}

.c1:focus {
  outline: none;
}

.c0 {
  z-index: 10;
  overflow: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  top: 0;
  left: 0;
  width: -webkit-fill-available;
  width: -moz-available;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 40px 0;
  box-sizing: border-box;
  background-color: var(--charcoal-surface4);
}

.c13 {
  position: absolute;
  top: 8px;
  right: 8px;
  color: var(--charcoal-text3);
  -webkit-transition: 0.2s color;
  transition: 0.2s color;
}

.c13:not(:disabled):not([aria-disabled]):hover,
.c13[aria-disabled='false']:hover {
  color: var(--charcoal-text3-hover);
}

.c13:not(:disabled):not([aria-disabled]):active,
.c13[aria-disabled='false']:active {
  color: var(--charcoal-text3-press);
}

.c3 {
  margin: 0;
  font-weight: inherit;
  font-size: inherit;
}

.c2 {
  height: 64px;
  display: grid;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  color: var(--charcoal-text1);
  font-size: 16px;
  line-height: 24px;
  font-weight: bold;
  display: flow-root;
}

.c4::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c4::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c5 {
  padding-bottom: 40px;
}

.c6 {
  border: solid 1px rgba(0,0,0,0.08);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 256px;
}

@media (max-width:743px) {
  .c1 {
    max-width: 440px;
    width: calc(100% - 48px);
    max-width: unset;
    width: 100%;
    border-radius: 0;
    margin: auto 0 0 0;
    min-height: 100%;
  }
}

@media (max-width:743px) {
  .c0 {
    padding: 0;
  }
}

@media (max-width:743px) {
  .c2 {
    height: 48px;
  }
}

<div
  data-dark={false}
>
  <div
    style={
      Object {
        "height": "10px",
      }
    }
  >
    <button
      onClick={[Function]}
    >
      open
    </button>
    <div
      className="c0"
      onPointerDown={[Function]}
      style={Object {}}
    >
      <div
        className="c1"
        onBlur={[Function]}
        onFocus={null}
        onKeyDown={[Function]}
        role="dialog"
        size="M"
        style={Object {}}
        tabIndex={-1}
      >
        <div
          className="c2"
        >
          <h3
            className="c3 c4"
          >
            modal
          </h3>
        </div>
        <div
          className="c5"
        >
          <div
            style={
              Object {
                "padding": "16px",
              }
            }
          >
            <div
              className="c6"
            >
              256px
            </div>
            <div
              className="c7"
            >
              <button
                className="c8"
                onClick={[Function]}
                type="button"
              >
                <span
                  className="c9"
                >
                  Apple
                </span>
                <pixiv-icon
                  class="c10"
                  name="16/Menu"
                />
              </button>
            </div>
            <div
              className="c6"
            >
              256px
            </div>
            <div
              className="c7"
            >
              <button
                className="c8"
                onClick={[Function]}
                type="button"
              >
                <span
                  className="c9"
                >
                  Apple
                </span>
                <pixiv-icon
                  class="c10"
                  name="16/Menu"
                />
              </button>
            </div>
            <div
              className="c6"
            >
              256px
            </div>
          </div>
        </div>
        <button
          className="c11 c12 c13"
          onClick={[Function]}
        >
          <pixiv-icon
            name="24/Close"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots DropdownSelector Playground 1`] = `
.c0 {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  padding: 0;
  border-style: none;
  outline: none;
  color: inherit;
  text-rendering: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  word-spacing: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: inherit;
  margin: 0;
  overflow: visible;
  text-transform: none;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c1 {
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  display: inline-grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  border-radius: 999999px;
  font-size: 14px;
  line-height: 22px;
  font-weight: bold;
  padding-right: 24px;
  padding-left: 24px;
  color: var(--charcoal-text2);
  background-color: var(--charcoal-surface3);
  -webkit-transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  height: 40px;
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active,
.c1:not(:disabled):not([aria-disabled]):focus,
.c1[aria-disabled='false']:focus,
.c1:not(:disabled):not([aria-disabled]):focus-visible,
.c1[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c1:not(:disabled):not([aria-disabled]):hover,
.c1[aria-disabled='false']:hover {
  color: var(--charcoal-text2-hover);
  background-color: var(--charcoal-surface3-hover);
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active {
  color: var(--charcoal-text2-press);
  background-color: var(--charcoal-surface3-press);
}

.c1:disabled,
.c1[aria-disabled]:not([aria-disabled='false']) {
  opacity: 0.32;
}

.c2 {
  display: inline-block;
  width: 100%;
}

.c2:disabled,
.c2[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
  opacity: 0.32;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  width: 100%;
  box-sizing: border-box;
  border: none;
  cursor: pointer;
  padding-right: 8px;
  padding-left: 8px;
  background-color: var(--charcoal-surface3);
  border-radius: 4px;
  -webkit-transition: 0.2s box-shadow;
  transition: 0.2s box-shadow;
}

.c3:disabled,
.c3[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c3:not(:disabled):not([aria-disabled]):focus,
.c3[aria-disabled='false']:focus,
.c3:not(:disabled):not([aria-disabled]):active,
.c3[aria-disabled='false']:active,
.c3:not(:disabled):not([aria-disabled]):focus-visible,
.c3[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c4 {
  text-align: left;
  font-size: 14px;
  line-height: 22px;
  display: flow-root;
  color: var(--charcoal-text2);
}

.c4::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c4::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c5 {
  color: var(--charcoal-text2);
}

<div
  data-dark={false}
>
  <div
    style={
      Object {
        "width": 288,
      }
    }
  >
    <button
      className="c0 c1"
      disabled={false}
    />
    <div
      className="c2"
    >
      <button
        className="c3"
        onClick={[Function]}
        type="button"
      >
        <span
          className="c4"
        >
          50
        </span>
        <pixiv-icon
          class="c5"
          name="16/Menu"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots DropdownSelector Section List 1`] = `
.c0 {
  display: inline-block;
  width: 100%;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
  opacity: 0.32;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  width: 100%;
  box-sizing: border-box;
  border: none;
  cursor: pointer;
  padding-right: 8px;
  padding-left: 8px;
  background-color: var(--charcoal-surface3);
  border-radius: 4px;
  -webkit-transition: 0.2s box-shadow;
  transition: 0.2s box-shadow;
}

.c1:disabled,
.c1[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c1:not(:disabled):not([aria-disabled]):focus,
.c1[aria-disabled='false']:focus,
.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active,
.c1:not(:disabled):not([aria-disabled]):focus-visible,
.c1[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c2 {
  text-align: left;
  font-size: 14px;
  line-height: 22px;
  display: flow-root;
  color: var(--charcoal-text2);
}

.c2::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c2::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c3 {
  color: var(--charcoal-text2);
}

<div
  data-dark={false}
>
  <div
    style={
      Object {
        "width": 288,
      }
    }
  >
    <div
      className="c0"
    >
      <button
        className="c1"
        onClick={[Function]}
        type="button"
      >
        <span
          className="c2"
        >
          Apple
        </span>
        <pixiv-icon
          class="c3"
          name="16/Menu"
        />
      </button>
    </div>
  </div>
</div>
`;
